frontend/catalyst/jax_primitives.py

```diff
@@ -730,10 +730,12 @@ def _qinst_lowering(
     name_str = str(name_attr)
     name_str = name_str.replace('"', "")
 
-    if name_str == "MultiRZ":
-        assert len(float_params) == 1, "MultiRZ takes one float parameter"
+    if name_str in {"MultiRZ", "GlobalPhase"}:
```

Comment:
Why is GlobalPhase a multi qubit op?

Reply:
Maybe this is a good opportunity to add it as another operation instead of modifying of MultiRZ due to it having "no regular wires".

Reply:
Although come to think of it that would completely eliminate an enforced ordering of the operation due to the lack of data dependencies 🤔 I guess we do want it acting on all wires at the MLIR level.

Reply:
Potentially reordered, but only if the op is marked side-effect free. But now I'm realizing this is actually a match better match to the underlying properties of the op. A global phase operation commutes with **any** other quantum operation. If it is controlled, then the only limiting factor on its commutation properties are the control wires, which are already represented in the IR.
The only ordering issue I see is the state measurement process (which is not a quantum operation), we would want to ensure the global phase op is not reordered with respect to the state instruction 🤔

Reply:
I was not able to create a global phase operation in JAX that took qubits but returned no qubits. This breaks several invariants in qreg. So, I just added the transform to make sure that it has at least some dependency on the wires and JAX doesn't reorder them. We can chat again tomorrow.

Reply:
> I was not able to create a global phase operation in JAX that took qubits but returned no qubits.
Oh why wouldn't it be returning qubits if accepts qubits in JAX?

Reply:
Because we don't want the MLIR operation to return Qubits. And JAX imposes a one-to-one correspondance when lowering to MLIR. If the operation returns qubits in JAX, it must in MLIR. This can be mitigated with an intermediate step that later removes the wires. But then, this essentially becomes the same as just keeping the current state and adding a transformation later.

Reply:
> Are you sure? The lowering function might need to return qubit values to JAX

I am talking specifically during the lowering of JAX to MLIR while it is still under JAX control. We need to return the MLIR values that correspond to JAX's return values.

Reply:
E.g.,

```
    return GradOp(
        flat_output_types,
        ir.StringAttr.get(method),
        ir.FlatSymbolRefAttr.get(symbol_name),
        mlir.flatten_lowering_ir_args(args_and_consts),
        diffArgIndices=diffArgIndices,
        finiteDiffParam=finiteDiffParam,
    ).results
```

The results match to JAX variables.

Reply:
Right, can't you just return the input values then? Not sure why it would impose restrictions on the actual operation.

Reply:
Maybe! That is something that had not occurred to me! Thanks!